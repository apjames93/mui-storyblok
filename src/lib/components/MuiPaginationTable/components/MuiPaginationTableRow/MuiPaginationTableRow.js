import React from 'react';
import PropTypes from 'prop-types';
import { TableRow } from '@material-ui/core';
import MuiTableCell from '../../../MuiTable/components/MuiTableRow/components/MuiTableCell/MuiTableCell';
import { validComponents } from '../../../../utils/customProps';
import { renderComponents } from '../../../../utils/customComponents';
import StoryBlok from '../../../../utils/Storyblok';

const MuiPaginationTableRow = ({
  rootClass,
  content,
  hover,
  selected,
}) => {
  const components = {
    MuiTableCell,
  };

  const styles = StoryBlok.arrayToMuiStyles(rootClass);

  return (
    <TableRow
      className={styles.root}
      hover={hover}
      selected={selected}
    >
      {content.map((component, key) => renderComponents(components, component, key))}
    </TableRow>
  );
};

export default MuiPaginationTableRow;

MuiPaginationTableRow.propTypes = {
  /**
   * stroyblok multiselect of css classes
   * Mui Override or extend the styles applied to the component.
   */
  rootClass: PropTypes.arrayOf(PropTypes.string),
  /**
   * mui prop: true | false
   * If true, the table row will shade on hover.
   */
  hover: PropTypes.bool,
  /**
   * mui prop: true | false
   * If true, the table row will have the selected shading.
   */
  selected: PropTypes.bool,
  /**
   * Content passed to render
   * components:
   * MuiTableCell
   */
  content(props, propName, componentName) {
    const components = ['MuiTableCell'];
    return validComponents(props, propName, componentName, components);
  },
};

MuiPaginationTableRow.defaultProps = {
  hover: false,
  selected: false,
  rootClass: [],
  content: [],
};
