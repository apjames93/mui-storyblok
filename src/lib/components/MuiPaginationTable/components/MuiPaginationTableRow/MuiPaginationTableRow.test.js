import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import MuiPaginationTableRow from './MuiPaginationTableRow';

function setup() {
  const props = {
    _editable: '<!--#storyblok#{"name": "MuiPaginationTableRow", "space": "48408", "uid": "947be9f0-47c3-4315-a95a-550f0c560eb5", "id": "307934"}-->',
    hover: false,
    selected: false,
    content: [{
      size: 'medium',
      align: 'center',
      content: [{
        align: 'inherit',
        color: 'initial',
        width: '100%',
        height: '100%',
        content: [{
          _editable: '<!--#storyblok#{"name": "MuiText", "space": "48408", "uid": "947be9f0-47c3-4315-a95a-550f0c560eb5", "id": "307934"}-->',
          text: 'Single table row',
          component: 'MuiText',
        }],
        variant: 'body1',
        _editable: '<!--#storyblok#{"name": "MuiTypography", "space": "48408", "uid": "947be9f0-47c3-4315-a95a-550f0c560eb5", "id": "307934"}-->',
        component: 'MuiTypography',
      }],
      padding: 'default',
      _editable: '<!--#storyblok#{"name": "MuiTableCell", "space": "48408", "uid": "947be9f0-47c3-4315-a95a-550f0c560eb5", "id": "307934"}-->',
      component: 'MuiTableCell',
      sortDirection: 'asc',
    }],
  };
  const comp = shallow(<MuiPaginationTableRow {...props} />);
  return { comp, props };
}

describe('<MuiPaginationTableRow />', () => {
  test('snapshot', () => {
    const { props } = setup();
    const tree = renderer.create(<MuiPaginationTableRow {...props} />);
    expect(tree).toMatchSnapshot();
  });

  it('should render MuiTable Row', () => {
    const { comp } = setup();
    expect(comp).toBeDefined();
  });
});
