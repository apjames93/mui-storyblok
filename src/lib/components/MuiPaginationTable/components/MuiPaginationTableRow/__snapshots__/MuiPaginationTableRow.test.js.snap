// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MuiPaginationTableRow /> snapshot 1`] = `
<tr
  className="MuiTableRow-root makeStyles-root-2"
  role={null}
>
  <td
    aria-sort="ascending"
    className="MuiTableCell-root makeStyles-root-3 MuiTableCell-alignCenter"
  >
    <div
      className="MuiTypography-root makeStyles-root-4 MuiTypography-body1"
    >
      <span
        className="MuiTypography-root makeStyles-root-5 MuiTypography-body1"
      >
         undefined 
      </span>
    </div>
  </td>
</tr>
`;
